zerver/lib/home.py

```diff
@@ -175,9 +175,13 @@ def build_page_params_for_home_page_load(
 
     furthest_read_time = get_furthest_read_time(user_profile)
 
+    default_language = register_ret["user_settings"]["default_language"]
+    if user_profile is None:
+        default_language = request.COOKIES.get(settings.LANGUAGE_COOKIE_NAME, default_language)
```

Comment:
I think there’s no need to call `get_and_set_request_language` in this case. If the correct language is stored in Django’s cookie, it will already have been set up by Django’s `LocaleMiddleware` and stored in `request.LANGUAGE_CODE`.

Reply:
Makes sense.